File: examples/java/core/wait/basic-wait.java
import java.time.Duration;
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;

public class BasicWaitExample extends DurableHandler<Object, String> {
    @Override
    public String handleRequest(Object input, DurableContext context) {
        // Wait for 5 seconds
        context.wait(null, Duration.ofSeconds(5));
        return "Wait completed";
    }
}

File: examples/csharp/core/wait/basic-wait.cs
using Amazon.Lambda.Core;
using Amazon.Lambda.DurableExecution;

public class BasicWaitExample
{
    public Task<DurableExecutionInvocationOutput> Handler(
        DurableExecutionInvocationInput input, ILambdaContext context)
        => DurableFunction.WrapAsync<object, string>(Workflow, input, context);

    private async Task<string> Workflow(object input, IDurableContext ctx)
    {
        // Wait for 5 seconds
        await ctx.WaitAsync(TimeSpan.FromSeconds(5));
        return "Wait completed";
    }
}
